Bugfix/boxplot model colors

mbwatson opened the pull request:
this PR brings in a few things. the primary goal was to address the model coloring, so i replicated in BoxplotChart how colors are applied in LineChart, which was pretty straightforward.

then i flipped the `disabled` attribute to `false` on the model controls--just for fun--and i was surprised to see that i could already select multiple models and see multiple boxplots within each subplot, so i went ahead and did these things:

1. attempted to address the cramped spacing--mostly between boxplots within single subplots (title style, placement, spacing, and stuff will come in another PR)--and for this:

  - i started with a `BOXPLOT_HEIGHT` variable and tried to build everything off of that.
  - it's not _quite_ right, but it does help make things a bit less cramped.

2. disabled the model options for which there is no associated data. i'm not sure my solution is the best long-term solution, or if there's a better, preferred, or existing alternative approach, but here's how that was implemented:

 - the `models_select` component was updated to accept a `data` parameter, allowing for dynamic generation of the options list.
 - in `visualization_editor`, the list of available models is determined by inspecting the dataframe of the chart.
 - a mapping from model IDs (present in the data) to model names (displayed in the UI) is performed to identify available models.
 - models that are not found in the data are `disabled` in the select dropdown.
 - this is only applied on the initial load of the explorer and does not dynamically update when other controls are changed.

### some screenshots

multiple models selected, stack layout
<img width="1193" height="1164" alt="Screenshot from 2025-12-17 09-38-53" src="https://github.com/user-attachments/assets/f6b9f9cd-238a-4a5c-8a94-01271cc033d4" />

multiple models selected, grid layout
<img width="1169" height="914" alt="Screenshot from 2025-12-17 09-38-59" src="https://github.com/user-attachments/assets/84fb400e-f3e6-4cf0-869d-0ef3b7922875" />

notice the model color alignment
<img width="1475" height="242" alt="image" src="https://github.com/user-attachments/assets/5e2f807b-0049-4961-8014-f6021065f530" />

disabled model options
<img width="516" height="391" alt="image" src="https://github.com/user-attachments/assets/e63dcb97-04eb-4ea6-ae06-82e00e4e91fc" />

mbwatson committed:
mirror model-coloring logic from LineChart class in BoxplotChart

mbwatson committed:
enable model selection control

mbwatson committed:
rethink dynamic box/subplot/chart height calculations

increaase subplot base height

disable models with no associated data

mbwatson committed:
address overlapping y-axis labels in boxplot grid layout

mbwatson committed:
add border-radius to chart container

mbwatson committed:
add spacing around chart title, subtitle & center

mbwatson committed:
increase time series chart left padding

davidborland reviewed (changes requested):
I think this is looking pretty good. 

One point of clarification:

So the models available in the dropdown will be those with data for whatever the visualization controls are when the explorer is loaded, e.g. if an insight was looking at North Carolina data, it would show models that have data for North Carolina, but that list wouldn't update if you switched to South Carolina?

I think this is an improvement over the current status, but we should create an issue to update dynamically based on the controls as they are adjusted.

Some changes I think we should make:

1. The legend and boxplot labels are redundant, so I would remove one of them. I would maybe lean toward keeping the labels and removing the legend, but I'm not sold on that.

2. We need the Scenario labels above each subplot.

mbwatson committed:
hide legend

mbwatson committed:
add scenario subplot titles

mbwatson commented:
> So the models available in the dropdown will be those with data for whatever the visualization controls are when the explorer is loaded...?

almost. the models available in the initial load are those that have _any_ records in our data, not just that of the initial controls, and that remains true even after controls are changed.

i agree there might be room for some nice dynamic updates to the options as the user makes other changes.

i removed the legend and added the subplot titles.

mbwatson committed:
enable searching in model select

davidborland reviewed (approved):
Looks good to me!

davidborland commented:
Actually, one more minor thing: maybe add just a little space between the model labels and the 0 line if easy to do, then feel free to merge.

mbwatson committed:
Merge branch 'enhancement/subplot-spacing' into bugfix/boxplot-model-colors

mbwatson commented:
> we should create an issue to update dynamically based on the controls as they are adjusted

just noting we have this issue #47
